shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+
+#include <Eigen/Core>
+#include <catch.hpp>
+#include <string>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/support/yaml_expression.hpp"
+
+
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+/**
+ * Unit tests for [utility][support][yaml_expression]
+ *
+ * @author Liam Craft
+ */
+
+
+SCENARIO("yaml nodes can be converted to a given container type", "[utility][support][yaml_expression]") {
+    GIVEN("A yaml node containing a list of data") {
+        YAML::Node testValues = YAML::LoadFile("/home/nubots/NUbots/shared/tests/data/yamlConversionTestVals.yaml");
+        // error threshold when comparing doubles
+        static constexpr double ERROR_THRESHOLD = 1e-6;
+        // ints
+        WHEN("A vector is declared using integers from the YAML node") {
+            std::vector<int> intVector = resolve_expression<int>(testValues["ints"]);
+            // Check that the values are good(container size, data type....)
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["ints"].size() == intVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                // Check values at each index
+                for (size_t i = 0; i < testValues["ints"].size(); i++) {
+                    REQUIRE(testValues["ints"][i].as<int>() == intVector.at(i));
+                }
+            }
+        }
+
+        // float
+        AND_WHEN("A vector is declared using float values from the YAML node") {
+            std::vector<float> floatVector = resolve_expression<float>(testValues["floats"]);
```

Comment:
snake_case pls
```suggestion
            std::vector<float> float_vector = resolve_expression<float>(test_values["floats"]);
```

Reply:
Force of habit. For the record though camelCase > snake_case.

Reply:
WRONG